### ISSUE TITLE
Installation with TensorFlow support failed on Mac OS

### ISSUE THREAD

**Issue by zarzen** [horovod/horovod#1468]
**Environment:**
1. Framework: TensorFlow
2. Framework version: 1.14
3. Horovod version: 0.18.2
4. MPI version: 4.0.2
5. CUDA version: X
6. NCCL version: X
7. Python version: 3.7.1
8. OS and version: Mac Catalina
9. GCC version: clang version 11.0.0

**Bug report:**
Installation through `HOROVOD_WITH_TENSORFLOW=1 pip install horovod` fails, because of following error:
```
-L/Users/zarzen/.pyenv/versions/3.7.1/lib/python3.7/site-packages/tensorflow -l:libtensorflow_framework.1.dylib
    ld: library not found for -l:libtensorflow_framework.1.dylib
```

I think the problem is due to the `-l` flag: `-l:libtensorflow_framework.1.dylib` should be `-ltensorflow_framework`. 

I changed the `setup.py` at [line249](https://github.com/horovod/horovod/blob/master/setup.py#L249) as following to get it successfully installed on my laptop. 
```
link_flags = ['-L/Users/zarzen/.pyenv/versions/3.7.1/lib/python3.7/site-packages/tensorflow', '-ltensorflow_framework']
        return tf.sysconfig.get_compile_flags(), link_flags
```

**Comment by tgaddair** [comment]
Hey @zarzen, this issue was caused by an upstream bug in TensorFlow 1.14 with custom ops: https://github.com/tensorflow/custom-op/issues/13.  A fix was landed here https://github.com/tensorflow/tensorflow/pull/30656 and should be available in 1.15 versions.

**Comment by zarzen** [comment]
Thanks!
